### PR TITLE
Automatic performance instrumentation for WebFlux

### LINKED ISSUE #1807: Automatic performance transactions for Webflux
Similar to SentryTracingFilter but for Webflux

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

This adds automatic performance instrumentation for WebFlux

## :bulb: Motivation and Context

Closes #1807 

## :green_heart: How did you test it?
Manually using samples + automated tests

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-samples/sentry-samples-spring-boot-webflux-jakarta/src/main/resources/application.properties [hunk 1]
```diff
@@ -8,3 +8,4 @@ sentry.max-breadcrumbs=150
 sentry.logging.minimum-event-level=info
 sentry.logging.minimum-breadcrumb-level=debug
 sentry.reactive.thread-local-accessor-enabled=true
+sentry.enable-tracing=false
```
Comment (sl0thentr0py) [inline sentry-samples/sentry-samples-spring-boot-webflux-jakarta/src/main/resources/application.properties:11]: why false here?

**Review by sl0thentr0py — APPROVED** [review]

File: sentry-samples/sentry-samples-spring-boot-webflux-jakarta/src/main/resources/application.properties (hunk 1 above)
Comment (adinauer) [reply sentry-samples/sentry-samples-spring-boot-webflux-jakarta/src/main/resources/application.properties]: Slipped in while testing; I changed it to `true`

**Comment by yhware** [comment]
Hi @adinauer 
Since this feature was added, I'm not sure which one I should use for tracing.
1. The one implemented here
2. Opentelemetry integration https://docs.sentry.io/platforms/java/performance/instrumentation/opentelemetry/

Do you guys have a recommendation?

**Comment by adinauer** [comment]
@yhware if you're already using OpenTelemetry, you'll probably want to use our OpenTelemetry Java Agent. Otherwise feel free to give this here a try first. If you have any feedback, feel free to share 🙏 .

**Comment by yhware** [comment]
Somehow the opentelemetry option is taking up a few hundred megs of memory and causing our dev containers to get OOMKilled.
But removing the opentelemetry option and updating sdk to the latest version does not give us performance metrics.
Is there a required spring/jdk version for it to work?

I enabled sentry.enable-tracing: true

**Comment by adinauer** [comment]
@yhware have you set sample rate to something < 1.0 or instrumenter to otel in your options? Otherwise try to enable debug, maybe that tells you something.

**Comment by adinauer** [comment]
We'll have to investigate memory consumption of the OTEL integration. Can you please share more details on what you did, like is it just simple HTTP request or something special? How long does it take to run OOM? What version of Spring (Boot) are you on?

**Pull request merged**